src/overlays/actors/ovl_Obj_Chan/z_obj_chan.c

```diff
@@ -31,49 +39,404 @@ const ActorInit Obj_Chan_InitVars = {
-static ColliderCylinderInit D_80BBAB30 = {
-    { COLTYPE_HARD, AT_NONE, AC_ON | AC_TYPE_PLAYER, OC1_NONE, OC2_TYPE_1, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK1, { 0x00000000, 0x00, 0x00 }, { 0xF7CFFFFF, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_NONE, },
+static ColliderCylinderInit sObjChanCylinderInit = {
+    {
+        COLTYPE_HARD,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_NONE,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK1,
+        { 0x00000000, 0x00, 0x00 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_NONE,
+    },
     { 48, 76, -60, { 0, 0, 0 } },
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80BBAB5C[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+void ObjChan_InitMain(ObjChan* this, GlobalContext* globalCtx); /* static */
+void ObjChan_InitPot(ObjChan* this, GlobalContext* globalCtx);  /* static */
+void func_80BBA488(ObjChan* this, GlobalContext* globalCtx);    /* static */
+void ObjChan_DrawPot(Actor* thisx, GlobalContext* globalCtx);   /* static */
+void func_80BBA930(ObjChan* this, GlobalContext* globalCtx);    /* static */
+
+static Vec3f sObjChanFlameSize[2] = { { 0.159999996424f, 0.109999999404f, 1.0f },
+                                      { 0.129999995232f, 0.0900000035763f, 1.0f } };
+static f32 sObjChanFlameYOffset[2] = { 1800, 1800 };
+static s32 sObjChanLoaded;
+
+void ObjChan_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s16 temp_t2;
+    ObjChan* this = THIS;
+
+    if (OBJCHAN_SUBTYPE(this) == OBJCHAN_SUBTYPE_MAIN) {
+        if (sObjChanLoaded) {
+            Actor_MarkForDeath(&this->actor);
+            return;
+        }
+        this->actor.room = -1;
+        sObjChanLoaded = true;
+    }
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sObjChanCylinderInit);
+    func_8013E3B8(&this->actor, this->cutscenes, ARRAY_COUNT(this->cutscenes));
+    switch (OBJCHAN_SUBTYPE(this)) {
+        case OBJCHAN_SUBTYPE_MAIN:
+            this->rotation = this->actor.shape.rot.y;
+            this->actor.shape.rot.y = 0;
+            ObjChan_InitMain(this, globalCtx);
+            break;
+        case OBJCHAN_SUBTYPE_POT:
+            this->actor.draw = ObjChan_DrawPot;
+            ObjChan_InitPot(this, globalCtx);
+            break;
+    }
+}
+
+void ObjChan_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjChan* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+u32 func_80BB9A1C(ObjChan* this, f32 arg1) {
+    f32 temp_f6;
+    f32 sp20;
+
+    sp20 = Math_SinS(this->unk1D4) * this->unk1D0;
+    temp_f6 = (Math_CosS(this->unk1D4) * 0.03834952f * this->unk1D0) + arg1;
```

Comment:
```suggestion
    f32 temp_f6 = (Math_CosS(this->unk1D4) * 0.03834952f * this->unk1D0) + arg1;
    f32 sp20 = Math_SinS(this->unk1D4) * this->unk1D0;
```
Would declaring them at the top like this work?